### ISSUE TITLE
Support shift-clicking a tab to close it. 

### ISSUE THREAD

**Issue by osi** [zed-industries/feedback#853]
### Check for existing issues

- [X] Completed

### Describe the feature

A feature from the Jetbrains suite of tools I enjoy is the ability to close editor tabs by clicking on them while holding down the shift key.

**Comment by rkusa** [comment]
Looks like you can already middle mouse click a tab to close it. This also seems to be the de facto standard for tab-based apps (e.g. in Sublime Text, VS Code, Firefox, Chrome, Windows Explorer, ...). I am not sure if we can rebind such UI keybindings yet though.

**Comment by JosephTLyons** [comment]
Yeah, I don't think we we have support yet to customize actions that involve mouse clicks.

**Comment by osi** [comment]
ah, thanks @rkusa ; i'm on a mac, where middle clicking isn't a standard interaction.
